domino_generator/example/ex2/generate_all.dart

```diff
@@ -1,6 +1,5 @@
 import 'package:domino_generator/src/component_generator.dart';
 
 void main() {
-  final gen = ComponentGenerator();
-  gen.compileDirectory('example/ex2/web-project');
+  compileDirectory('example/ex2/web-project');
```

Comment:
I think keeping the `ComponentGenerator` has an advantage: it can be the place to registering multiple namespaces (e.g. similar to XML namespaces, a component namespace alias should be specified by its user, and not the schema). It can still use the `TemplateRegistry` internally, but it probably should have the functionality of the `compileDirectory` as a utility method on `ComponentGenerator` (which can be preconfigured with some defaults).